### ISSUE TITLE
`deno publish --dry-run` does not have a non-zero exit code if there are non-supported HTTP imports but no slow types

### ISSUE THREAD

**Issue by roj1512** [denoland/deno#23023]
Version: Deno 1.41.3

**Comment by iuioiua** [comment]
I was unable to reproduce this issue with a minimal setup. Are you able to provide the source code for reproduction? Also, have you tried testing this on canary?
```jsonc
// deno.json
{
  "name": "@iuioiua/hello",
  "version": "1.0.0",
  "exports": "./mod.ts"
}
```
```ts
// mod.ts
import { assert } from "https://deno.land/std@0.220.1/assert/assert.ts";
assert("hello".includes("ello"));
```

**Comment by roj1512** [comment]
@iuioiua On 1.42.0, try:

```
git clone https://github.com/MTKruto/MTKruto
cd MTKruto
git checkout 1deda5c978faefa481e9b84e0fb59c799eb7a772

# The below only modifies Deno config.
deno run -A https://raw.githubusercontent.com/MTKruto/tools/4ca0c0207650af687fc5f4e550639910f2cc99f8/pre_jsr.ts

# Fails even though there are HTTP imports in 0_deps.ts.
deno publish --dry-run --allow-dirty
```

**Comment by roj1512** [comment]
@iuioiua On 1.42.0, try:

```shell
git clone https://github.com/MTKruto/MTKruto
cd MTKruto
git checkout 1deda5c978faefa481e9b84e0fb59c799eb7a772

# The below only modifies Deno config.
deno run -A https://raw.githubusercontent.com/MTKruto/tools/4ca0c0207650af687fc5f4e550639910f2cc99f8/pre_jsr.ts

# Succeeds even though there are HTTP imports in 0_deps.ts.
deno publish --dry-run --allow-dirty
```

**Comment by iuioiua** [comment]
I'm able to reproduce this issue. Findings so far: adding a slow type to `mod.ts` adds the following files to the module graph when running `deno publish --dry-run`, which should've been there to begin with:
```

file:///Users/asher/GitHub/MTKruto/utilities/1_base64.ts
file:///Users/asher/GitHub/MTKruto/0_deps.ts
https://deno.land/std@0.221.0/assert/mod.ts
https://deno.land/std@0.221.0/path/mod.ts
https://deno.land/std@0.221.0/encoding/base64.ts
https://deno.land/std@0.221.0/media_types/content_type.ts
https://deno.land/std@0.221.0/media_types/extension.ts
https://deno.land/x/tgcrypto@0.3.3/mod.ts
https://raw.githubusercontent.com/MTKruto/compress/main/mod.ts
https://deno.land/x/html_parser@v0.1.3/src/mod.ts
file:///Users/asher/GitHub/MTKruto/utilities/0_buffer.ts

file:///Users/asher/GitHub/MTKruto/utilities/0_control.ts

file:///Users/asher/GitHub/MTKruto/utilities/0_hash.ts

file:///Users/asher/GitHub/MTKruto/utilities/0_mutex.ts
file:///Users/asher/GitHub/MTKruto/utilities/0_object.ts
file:///Users/asher/GitHub/MTKruto/utilities/0_rle.ts


file:///Users/asher/GitHub/MTKruto/utilities/1_auth.ts
file:///Users/asher/GitHub/MTKruto/utilities/1_misc.ts
file:///Users/asher/GitHub/MTKruto/utilities/1_queue.ts

file:///Users/asher/GitHub/MTKruto/transport/0_obfuscation.ts

file:///Users/asher/GitHub/MTKruto/tl/0_tl_raw_writer.ts

file:///Users/asher/GitHub/MTKruto/tl/3_deserialize.ts

file:///Users/asher/GitHub/MTKruto/client/0_message.ts

file:///Users/asher/GitHub/MTKruto/client/0_types.ts


file:///Users/asher/GitHub/MTKruto/client/0_utilities.ts
file:///Users/asher/GitHub/MTKruto/client/1_account_manager.ts
file:///Users/asher/GitHub/MTKruto/client/1_bot_info_manager.ts

```

**Comment by roj1512** [comment]
No longer reproducing this with canary.

**Comment by iuioiua** [comment]
Yes, this issue was fixed in #23187. I'm working on a regression test for this in #23369. Either way, thank you for reporting and confirming.
